corehq/messaging/scheduling/static/scheduling/js/conditional_alert_main.js

```diff
@@ -56,20 +56,16 @@ hqDefine(
                 // compare changes with initialFormContent to check if only 'message' has changed
                 var formContentChanges = [];
                 Object.keys(finalFormContent).forEach(function (key) {
-                    finalFormContent[key] !== initialFormContent[key]
-                        ? formContentChanges.push(key)
-                        : null;
+                    if (finalFormContent[key] !== initialFormContent[key]) {
+                        formContentChanges.push(key);
+                    }
                 });
                 var runMessagingRule = !_.isEqual(
-                    ["message"],
+                    ["content-message"],
                     formContentChanges
                 );
                 // to convey to HQ to reprocess or not
-                var input = $("<input />")
-                    .attr("type", "hidden")
-                    .attr("name", "runMessagingRule")
-                    .attr("value", runMessagingRule);
-                $(this).append(input);
+                $("#should_run_rule").val(runMessagingRule);
```

Comment:
:+1:

Reply:
also can this be
```
if (_.isEqual(["content-message"], formContentChanges)) {
    $("#should_run_rule").val(false)
}
```
else it would stay as default true